### PR TITLE
Remove IndexHandle

### PR DESCRIPTION
**Pull request by graetzer**
### Scope & Purpose

Remove useless index handle code

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)

### Testing & Verification

This change is a trivial rework / code cleanup without any test coverage.

http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/7155/

### CODE REVIEW

**Comment by jsteemann** [comment]
I was always wondering what this `IndexHandle` thing was for.

File: lib/Futures/SharedState.h [hunk 1]
```diff
@@ -155,25 +155,23 @@ class SharedState {
     _callback = std::forward<F>(func);
 
     auto state = _state.load(std::memory_order_acquire);
-    while (true) {
```
Comment (danielhlarkin) [inline lib/Futures/SharedState.h]: Could you comment on why this seemingly unrelated change here?

**Comment by jsteemann** [comment]
https://172.16.10.101/view/PR/job/arangodb-matrix-pr/7173/

File: lib/Futures/SharedState.h (hunk 1 above)
Comment (jsteemann) [reply lib/Futures/SharedState.h]: The `while` loop seems unnecessary here.
In the code before this change, the while loop can only be repeated if one had also hit the `TRI_ASSERT(false);` assertion directly before. That means in maintainer mode, there will be at most a single loop iteration. In release mode however, there could have been multiple loop iterations in theory, because the assert will not trigger.
In reality, execution should never reach the `TRI_ASSERT(false);` location here. We can still leave it in place for debugging, but as it is not supposed to be reached, we don't need to do anything special for release builds, and can still remove the while loop here.
